### PR TITLE
Add execute phase to dynamic analysis

### LINKED ISSUE #844: Create new dynamic analysis phase for code execution
#730 was implemented as part of the import phase, but for detection reasons, it would be better to have it as a separate phase. 

This involves:

1. Adding a new dynamic analysis run phase
2. Adjusting the JSON schema for dynamic analysis. The results for each phase should probably be represented as a list of repeated structs rather than multiple / duplicate nested structs
3. migrating existing JSON dynamic analysis data over to the new format

### PR DESCRIPTION
**Pull request by maxfisher-g**
Fixes #844

### CODE REVIEW

File: internal/worker/code_execution.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	if err := sb.CopyIntoSandbox(ctx, tempPath, sandboxExecutionLogPath); err != nil {
+		return fmt.Errorf("could not copy empty execution log file to sandbox: %w", err)
+	}
+
+	// file wasn't written to, so don't worry too much about remove errors
+	_ = os.Remove(tempPath)
+
+	return nil
+}
```
Comment (calebbrown) [inline internal/worker/code_execution.go:54]: Is this function, and the use of `execution.log` necessary any more for gating the logic?

Given that the phase is now explicit I suspect this capability is no longer needed.

File: internal/worker/rundynamic.go [hunk 2]
```diff
@@ -230,18 +126,50 @@ func RunDynamicAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, sbOpts []sandb
...
-		return result, nil
+	return result, nil
+}
+
+func runDynamicAnalysisPhase(ctx context.Context, pkg *pkgmanager.Pkg, sb sandbox.Sandbox, analysisCmd string, phase analysisrun.DynamicPhase, result *DynamicAnalysisResult) error {
+	if phase == analysisrun.DynamicPhaseExecute {
+		// silently skip execute phase if disabled globally or for specific ecosystem
+		if !isExecutePhaseEnabled(pkg.Ecosystem()) {
+			return nil
+		}
```
Comment (calebbrown) [inline internal/worker/rundynamic.go:137]: This logic could be avoided by wrapping `analysisrun.DefaultDynamicPhases()` in here and having it return the correct set of phases based on the `pkg.Ecosystem` and the feature flag.

**Review by calebbrown — CHANGES_REQUESTED** [review]

File: internal/worker/code_execution.go (hunk 1 above)
Comment (maxfisher-g) [reply internal/worker/code_execution.go:54]: Yes it can be removed.

File: internal/worker/rundynamic.go (hunk 2 above)
Comment (maxfisher-g) [reply internal/worker/rundynamic.go]: Done

File: sandboxes/dynamicanalysis/analyze-python.py [hunk 3]
```diff
@@ -85,33 +86,46 @@ def import_single_module(import_path):
 
 def import_module(import_path):
     print('Importing', import_path)
+    # noinspection PyBroadException
     try:
-        module = importlib.import_module(import_path)
-    except:
+        importlib.import_module(import_path)
+    except Exception:
```
Comment (calebbrown) [inline sandboxes/dynamicanalysis/analyze-python.py:92]: This should probably still be a bare `except:` to prevent the script from aborting unexpectedly.

**Review by calebbrown — APPROVED** [review]

File: sandboxes/dynamicanalysis/analyze-python.py (hunk 3 above)
Comment (maxfisher-g) [reply sandboxes/dynamicanalysis/analyze-python.py:92]: Okay. Bare `except` is equivalent to `except BaseException` so I'll put that to make it more explicit, and add a comment

**Pull request merged**